doc/tutorials/lattice_boltzmann/lattice_boltzmann_sedimentation.ipynb

```diff
@@ -0,0 +1,396 @@
+   "source": [
+    "We will set an external force acting on all particles. You can take this to model the effect of gravity, for example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "f_gravity = [0, -3, 0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we start with sampling using the Langevin thermostat (with temperature $T=0$).\n",
+    "\n",
+    "**possible tasks: set Langevin thermostat, add particles**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Langevin simulation\n",
+    "system.thermostat.set_langevin(kT=0., gamma=15, seed=12)\n",
+    "\n",
+    "parts = system.part.add(pos=initial_positions, ext_force=[f_gravity] * n_parts)\n",
+    "\n",
+    "system.integrator.run(0)\n",
+    "\n",
+    "for step in tqdm(range(sampling_steps)):\n",
+    "    data_lv[step, :] = parts.pos_folded    \n",
+    "    system.integrator.run(25)\n",
+    "data_lv[-1, :] = parts.pos_folded\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we want to sample the same system, but with a coupled lattice-Boltzmann fluid. We first reset the particles to their initial positions and remove any particle velocities.\n",
+    "Then we set up the LB fluid. The wall constraints that were previously added have to be also registered as LB boundaries.\n",
+    "\n",
+    "**possible tasks: set up LB fluid, add lbboundaries**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# LB simulation cleanup and setup\n",
+    "parts.pos = initial_positions\n",
+    "parts.v = [0, 0, 0]\n",
+    "system.thermostat.turn_off()\n",
+    "\n",
+    "lbf = espressomd.lb.LBFluid(agrid=spacing, dens=1.0, visc=1.0, tau=system.time_step, kT=0)\n",
+    "system.actors.add(lbf)\n",
+    "system.thermostat.set_lb(LB_fluid=lbf, seed=123, gamma=15)\n",
+    "\n",
+    "# add LB boundaries at walls\n",
+    "for wall_shape in [wall_shape_b, wall_shape_t]:\n",
+    "    no_slip_wall = espressomd.lbboundaries.LBBoundary(\n",
+    "        shape=wall_shape, velocity=[0, 0, 0])\n",
+    "    system.lbboundaries.add(no_slip_wall)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# numpy array to hold flowfield data\n",
+    "data_flowfield = np.full((sampling_steps, n_height, n_width, 3), np.nan)\n",
+    "\n",
+    "# return n_x x n_x array of lattice velocities, averaged in z-direction \n",
+    "def mean_flowfield(xs=n_width, ys=n_height, zs=n_depth):\n",
+    "    dataframe = np.array([[[lbf[x, y, z].velocity for z in range(zs)] for x in range(xs)] \n",
+    "                          for y in range(ys)])\n",
+    "    return np.mean(dataframe, axis=2)\n",
```

Comment:
Consider using a LB profile observable here, a particle position observable and a particle velocity observable, and accumulators. This way we don't need to re-implement core features as numpy operations, and can remove the numpy arrays pre-filled with NaN values.